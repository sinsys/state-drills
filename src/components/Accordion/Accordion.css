.Accordion {
	list-style: none;
	max-width: 400px;
}
.Accordion li {
	width: 100%;
}