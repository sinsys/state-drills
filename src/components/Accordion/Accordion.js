import React from 'react';
import './Accordion.css';

class Accordion extends React.Component {
	static defaultProps = {
		sections: []
	};

	constructor(props) {
		super(props);
		this.state = {
			activeSection: null
		};
	};

	renderSection(section, id, activeSection) {
		return (
			<li
				className="Accordion-section"
				key={id}
			>
				<button
					type="button"
					onClick={() => {
						this.handleSectionSelect(id)
					}}
				>
					{section.title}
				</button>
				{(activeSection === id) &&
					<p>
						{section.content}
					</p>
				}
			</li>
		)
	}

	handleSectionSelect = (index) => {
		this.setState({
			activeSection: index
		})
	};

	render() {
		const { activeSection } = this.state;
		const { sections } = this.props;

		return (
			<div className="Accordion-wrapper">
				<h2>Accordion</h2>
				<ul className="Accordion">
					{sections.map((section, id) => (
						this.renderSection(section, id, activeSection)
					))}
				</ul>
			</div>
		)
	};
}

export default Accordion;