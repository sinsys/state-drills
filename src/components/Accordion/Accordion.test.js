import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const sectionsProp = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]



describe("Accordion component", () => {

	const wrapper = shallow(<Accordion sections={sectionsProp} />);
	const div = document.createElement('div');

	it('renders without errors', () => {
		ReactDOM.render(<Accordion />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders empty with no sections by default', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	})

	it('opens any clicked section', () => {
    const singleItem = wrapper.find('button').at(1).simulate('click').simulate('click');
    expect(toJson(singleItem)).toMatchSnapshot();
  })

	it('opens one section at a time', () => {
		wrapper.find('button').forEach((button, index) => {
			button.simulate('click');
		});
		expect(toJson(wrapper)).toMatchSnapshot();
	});

});