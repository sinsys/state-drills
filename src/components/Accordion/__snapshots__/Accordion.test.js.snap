// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component opens any clicked section 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Section 2
</button>
`;

exports[`Accordion component opens one section at a time 1`] = `
<div
  className="Accordion-wrapper"
>
  <h2>
    Accordion
  </h2>
  <ul
    className="Accordion"
  >
    <li
      className="Accordion-section"
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      className="Accordion-section"
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      className="Accordion-section"
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
      <p>
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordion component renders empty with no sections by default 1`] = `
<div
  className="Accordion-wrapper"
>
  <h2>
    Accordion
  </h2>
  <ul
    className="Accordion"
  >
    <li
      className="Accordion-section"
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
      </button>
    </li>
    <li
      className="Accordion-section"
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
      </button>
    </li>
    <li
      className="Accordion-section"
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
